distribution/lib/Standard/Database/0.0.0-dev/src/Internal/IR/Query.enso

```diff
@@ -62,3 +62,22 @@ type Query
          The columns in that query should correspond to the columns in specified
          in `column_names`, matching by position.
     Insert_From_Select (table_name:Text) (column_names : Vector Text) (select:Query)
+
+    ## PRIVATE
+       An SQL UPDATE query that updates rows in the target table with values
+       from the source table when the key columns match. Target table rows that
+       do not match the source are left unaffected.
+
+       This will usually be a query of the form
+       `UPDATE target SET c1=source.c1, ... FROM source WHERE target.key1=source.key1 AND ...`
+       where `c1` comes from `column_names` and `key1` comes from `key_columns`.
+    Update_From_Table (target_table_name:Text) (source_table_name:Text) (column_names : Vector Text) (key_columns : Vector Text)
+
+    ## PRIVATE
+       An SQL DELETE query that deletes from target table rows that are _not_
+       present in the source table, based on the specified key columns.
+       They key columns must be present under the same name in both tables.
+
+       This will usually be a query of the form
+       `DELETE FROM target WHERE (key_columns...) NOT IN (SELECT key_columns... FROM source)`.
```

Comment:
I wonder if a DELETE/JOIN would work better. 

Add a constant marker to the source, and then something like:

```sql
DELETE target
  FROM target 
  LEFT JOIN source ON  ...
 WHERE source.marker is NULL
```

Reply:
<img width="751" alt="image" src="https://github.com/enso-org/enso/assets/1436948/8f11c7a3-4404-4186-a855-b2bc5ee20652">
<img width="744" alt="image" src="https://github.com/enso-org/enso/assets/1436948/299ac29b-7c93-4827-a3f1-e95609384ef8">

This kind of syntax seems to not be supported in neither of the DBs we currently have.

Reply:
In SQLite the `DELETE` syntax only allows for a `WHERE` clause, no joins are available.
<img width="467" alt="image" src="https://github.com/enso-org/enso/assets/1436948/4b8183e8-b137-4ae7-aa86-23a1fdd8d417">

In Postgres, I guess we may try to play with `MERGE` but that is for a later ticket, I think - so to be revisited in #7036